src/dict.c

```diff
@@ -870,136 +869,163 @@ void dictTwoPhaseUnlinkFree(dict *d, dictEntry *he, dictEntry **plink, int table
+    }
+
+#define DICT_SET_VALUE(e, f, v)                                                                                        \
+    {                                                                                                                  \
+        if (entryIsNormal(e)) {                                                                                        \
+            _dictEntryNormal *_de = decodeEntryNormal(e);                                                              \
+            _de->f = v;                                                                                                \
+        } else if (entryIsEmbedded(e)) {                                                                               \
+            _dictEntryEmbedded *_de = decodeEntryEmbedded(e);                                                          \
+            _de->f = v;                                                                                                \
+        } else {                                                                                                       \
+            assert(0);                                                                                                 \
+        }                                                                                                              \
+    }
+
+#define DICT_INCR_VALUE(e, f, v)                                                                                       \
+    {                                                                                                                  \
+        if (entryIsNormal(e)) {                                                                                        \
+            _dictEntryNormal *_de = decodeEntryNormal(e);                                                              \
+            _de->f += v;                                                                                               \
+        } else if (entryIsEmbedded(e)) {                                                                               \
+            _dictEntryEmbedded *_de = decodeEntryEmbedded(e);                                                          \
+            _de->f += v;                                                                                               \
+        } else {                                                                                                       \
+            assert(0);                                                                                                 \
+        }                                                                                                              \
+    }
+
+#define DICT_GET_VALUE(e, f)                                                                                           \
+    (assert(entryHasValue(e)), entryIsNormal(e) ? decodeEntryNormal(e)->f : decodeEntryEmbedded(e)->f)
```

Comment:
Could we write the assertion and return statement multiple lines like the previous method? Was easier to follow.

Reply:
Unfortunately, that won't work. We need an expression here, as opposed to statements.